app/components/modals/motion_modal.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Modals
+  # Modal controlled by motion
+  class MotionModal < ViewComponent::Base
+    include Motion::Component
+
+    attr_reader :selected, :content, :show
+
+    map_motion :dismiss
+
+    def initialize(on_dismiss:, selected:, content:)
+      @on_dismiss = on_dismiss
+      @selected = selected
+      @content = content
+      @show = false
+
+      every 0.1.second, :fade_in
+    end
+
+    ## Periodic timers
+    # The periodic timer allows the background to animate in and out
+    # If you didn't care about the css animation, you could remove them and
+    # broadcast in the `dismiss` method
+    def fade_in
+      @show = true
+
+      stop_periodic_timer :fade_in
+    end
+
+    def fade_out
+      # This actually closes the modal, 0.1s later
+      @on_dismiss.call({event: "dismiss-modal"})
+
+      stop_periodic_timer :fade_out
+    end
+    ## End Periodic timers
+
+    ## Map motion events
+    def dismiss(event)
+      # Did they click inside the modal, or click a button or outside modal to close
+      return unless event.target.data["value"] == "dismissable"
+
```

Comment:
i guess maybe everything inside the modal element are sending events? this looks like the piece i was missing. i guess i didn't understand how the events were being sent to the server.

Reply:
Yes, everything inside the div is working with motion.